### DISCUSSION TITLE
Queue로 들어온 요청 결과를 Service에 반환

### DISCUSSION THREAD

**Discussion by Lcrescent3es** [100-hours-a-week/discussions#102] [개발] [answered]
목업 데이터가 부족해서 사용자가 외부 api를 호출해야하는 상황입니다.
외부 api에는 초당 20회의 제한이 있어, RestClient에 Rate Limiter를 적용해 Bean으로 선언하려고 합니다.
하지만, 외부 api를 크론잡으로도 활용하고 있어 사용자의 요청이 우선적으로 처리되도록 작업 우선순위를 설정하고자 합니다.

계획은 아래와 같습니다.

<img width="604" alt="image" src="https://github.com/user-attachments/assets/c6bf2109-db99-4fa1-82b5-9d6eefdd2ac5">

queue는 Spring AMQP와 Rabbit MQ를 활용해 구현하려고 합니다.
여기서, 호출한 Service에 외부 api 결과를 반환할 방법을 모르겠습니다.
RestClient 워커에 콜백 함수를 넘겨주거나, 워커 자체적으로 요청별 메서드를 호출해 처리하는 방법도 고민해봤지만, 정확한 방법이나 키워드가 없습니다.
조언해주시면 감사하겠습니다!

**Comment by ej31** [comment, marked as answer]
여러가지 방법이 있겠으나 제 생각에는..

1. 요청에 대한 응답 값에 DB 에서 조회할 수 있는 ID를 내려주면 됩니다.
2. 그리고 프론트에서는 폴링 방식으로 데이터 들어 왔는지 주기적으로 확인하면 됩니다.
3. SSE 방식도 적용해 볼 수 있겠으나 굳이..? 라고 생각이 들긴합니다.

**Reply by ej31** [reply]
Endpoint 가 POST, GET 두개를 만들면 되는것이지요
